File: Easy/Javascript/1-Two Sum.js
/**
 * first we create a map of indices so we don't have to loop again
 * then we find the difference of the target and current number
 * if the difference is available in the array we return the indices
 * of them both
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
let twoSum = function(nums, target) {
    const numIndices = {};

    for (let i = 0; i < nums.length; i++) {
        let currentNum = nums[i];
        let missing = target - currentNum;
        if (numIndices.hasOwnProperty(missing)) {
            return [numIndices[missing], i];
        }

        numIndices[currentNum] = i;
    }
};

console.log(twoSum([3,3],6))

File: Easy/Python/1-Two sum.py
'''
In this question i looped over every single element and finds out
what's the difference between the target and i if the difference
is available in the array return the indexx of them both 
'''
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        length = len(nums)
        for i in range(length):
            missing = target - nums[i]
            if missing in nums[i+1:]:
                return[i,nums.index(missing,i+1)]

print(Solution().twoSum([3,3],6))
